src/SFML/Audio/SoundFileReaderFlac.cpp

```diff
@@ -267,7 +261,21 @@ void SoundFileReaderFlac::seek(Uint64 sampleOffset)
     m_clientData.leftovers.clear();
 
     // FLAC decoder expects absolute sample offset, so we take the channel count out
-    FLAC__stream_decoder_seek_absolute(m_decoder, sampleOffset / m_channelCount);
+    if (sampleOffset < m_clientData.info.sampleCount)
+    {
+        // The "write" callback will populate the leftovers buffer with the first batch of samples from the
+        // seek destination, and since we want that data in this typical case, we don't re-clear it afterward
+        FLAC__stream_decoder_seek_absolute(m_decoder, sampleOffset / m_clientData.info.channelCount);
```

Comment:
Is this purely for readability? The result of that division op is only used once in the if case and once in the mutually-exclusive else case. It's never repeated in any single call to `seek()`, so I can't see it optimizing anything.